### PR TITLE
Update ci.yml to use a shared maven-ci-yml

### PR DESCRIPTION
**Pull request by olamy**

### CODE REVIEW

**Comment by olamy** [comment]
@joakime I hope you will like that. so we will have to maintain only the shared maven-ci.yml file with this

**Comment by joakime** [comment]
This latest build / action has several entries like `Maven CI / Maven Build / ${{ matrix.os }} jdk-${{ matrix.jdk }}-${{ matrix.distribution }}-mvn-${{ inputs.maven...`

The concept I like, the practical usages I'm not so sure.
We have some divergent use cases across github.com/jetty-project 
At least this is reversible on a per-sub-project basis.

Fix that odd naming issue on the actions, and we'll give it a whirl.

**Comment by olamy** [comment]
> This latest build / action has several entries like `Maven CI / Maven Build / ${{ matrix.os }} jdk-${{ matrix.jdk }}-${{ matrix.distribution }}-mvn-${{ inputs.maven...`
> 
> The concept I like, the practical usages I'm not so sure. We have some divergent use cases across github.com/jetty-project At least this is reversible on a per-sub-project basis.
> 

they all use a maven build with possible different jdk matrix, mvn goals to run is an option as well, eventually os.
not sure what is different between projects? 
we can avoid to waste time updating some gh actions everywhere and have only a single place.

> Fix that odd naming issue on the actions, and we'll give it a whirl.

yup done

**Comment by joakime** [comment]
This current PR builds both on `pull_request` and `push`. 
We need to fix that, as it causes double builds.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,35 +1,13 @@
-name: GitHub CI
+name: Maven Build CI
 
-on: [push, pull_request]
+on:
+  push:
```
Comment (joakime) [inline .github/workflows/ci.yml:4]: either drop `push`, or limit it to just `master` branch.

aka

``` yml
on:
  pull_request
push:
  branches:
    - master
```

**Comment by joakime** [comment]
Looking a lot better.
Merge it.

**Comment by joakime** [comment]
Squash Merge please.

**Pull request merged**